src/main/java/com/fidesmo/tutorials/hellofidesmo/MainActivity.java

```diff
@@ -100,42 +102,41 @@ protected void onNewIntent(Intent intent) {
 
     @Override
     public void tagDiscovered(Tag tag) {
-        Log.i(TAG, "Card detected on the NFC interface!");
+        setMainMessage(R.string.reading_card);
         try {
-            setMainMessage(R.string.reading_card);
             IsoCard isoCard = AndroidCard.get(tag);
-            readCard(isoCard);
-        } catch(IOException ioe) {
-            Log.e(TAG, "Failed to produce card from tag", ioe);
+            communicateWithCard(isoCard);
+        } catch(IOException e) {
+            e.printStackTrace();
         }
     }
 
     /**
      * Sends a SELECT APDU to the HelloFidesmo cardlet on the card and parses the response
-     * - If the response's status bytes are '90 00' (=APDU successfully executed), it displays the response payload
+     * - If the response's status bytes are '90 00' (=APDU successfully executed, Apdu.OK_APDU), it displays the response payload
      * - If not, it assumes that HelloFidesmo cardlet was not installed and shows a button
      *   so the user can launch the installation process
      * @param isoCard card detected at the NFC interface, supporting ISO 14443/4 standard
      */
-    protected void readCard(final IsoCard isoCard) {
-        byte[] response = null;
+    private void communicateWithCard(IsoCard isoCard) {
         try {
             isoCard.connect();
-            response = isoCard.transceive(Apdu.select(APPLICATION_ID, APP_VERSION));
+            //This is where you use your appId to select your app on the card (the one assigned to you when signing up to the Dev Portal)
+            byte[] response = isoCard.transceive(Apdu.select(APPLICATION_ID, APP_VERSION));
 
-            // Analyze the response. Its last two bytes are the status bytes - '90 00' means 'success'
+            // Analyze the response. Its last two bytes are the status bytes - '90 00'/Apdu.OK_APDU means 'success'
             if (Apdu.hasStatus(response, Apdu.OK_APDU)) {
-                Log.i(TAG, "Card returned SUCCESS");
+                setMainMessage("Select OK");
```

Comment:
I'd define a new string `select_ok`  in strings.xml, to be consistent.
Same for `select_not_ok` below